tests/vcs/test_clone.py

```diff
@@ -67,55 +67,87 @@ def test_clone_should_abort_if_user_does_not_want_to_reclone(mocker, tmpdir):
 
 
 @pytest.mark.parametrize(
-    'repo_type, repo_url, repo_name',
+    'repo_class, input_url, repo_url, repo_name',
     [
-        ('git', 'https://github.com/hello/world.git', 'world'),
-        ('hg', 'https://bitbucket.org/foo/bar', 'bar'),
-        ('git', 'git@host:gitoliterepo', 'gitoliterepo'),
-        ('git', 'git@gitlab.com:cookiecutter/cookiecutter.git', 'cookiecutter'),
-        ('git', 'git@github.com:cookiecutter/cookiecutter.git', 'cookiecutter'),
+        (vcs.Git, 'https://github.com/hello/world.git', None, 'world'),
+        (vcs.Hg, 'https://bitbucket.org/foo/bar', None, 'bar'),
+        (vcs.Git, 'git@host:gitoliterepo', 'git@host:gitoliterepo', 'gitoliterepo'),
+        (vcs.Git, 'git@gitlab.com:cookiecutter/cookiecutter.git', None, 'cookiecutter'),
+        (vcs.Git, 'git@github.com:cookiecutter/cookiecutter.git', None, 'cookiecutter'),
+        (
+            vcs.SVN,
+            'svn+https://private.com/myrepo',
+            'https://private.com/myrepo',
+            'myrepo',
+        ),
+        (vcs.SVN, 'svn://private.com/myrepo', None, 'myrepo'),
+        (vcs.SVN, 'svn+ssh://private.com/myrepo', None, 'myrepo'),
     ],
 )
 def test_clone_should_invoke_vcs_command(
-    mocker, clone_dir, repo_type, repo_url, repo_name
+    mocker, clone_dir, repo_class, input_url, repo_url, repo_name
 ):
     """When `clone()` is called with a git/hg repo, the corresponding VCS \
     command should be run via `subprocess.check_output()`.
 
     This should take place:
     * In the correct dir
     * With the correct args.
+
+    :param repo_class: _VCS class that should be used
+    :param input_url: Repo URL to be passed into the vcs.clone() function
+    :param repo_url: Repo URL that should be cloned by the VCS
+      (defaults to input_url if set to None)
+    :param repo_name: Name of subdirectory where the cloned files will be
     """
-    mocker.patch('cookiecutter.vcs.is_vcs_installed', autospec=True, return_value=True)
+    mocker.patch(
+        repo_class.__module__ + '.' + repo_class.__name__ + '.is_installed',
+        autospec=True,
+        return_value=True,
+    )
 
     mock_subprocess = mocker.patch(
         'cookiecutter.vcs.subprocess.check_output', autospec=True,
     )
     expected_repo_dir = os.path.normpath(os.path.join(clone_dir, repo_name))
 
+    # repo_url defaults to input_url
+    if repo_url is None:
+        repo_url = input_url
```

Comment:
I think it would be better to just write two times the same url in the `@pytest.mark.parametrize(` statement in those cases instead of this logic. In this way it's less clear that this is due to the particularity of svn that doesn't wants the `svn+` in the repo url.